src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -368,13 +368,9 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
             {
                 if (isRMW)
                 {
-                    if (targetReg != op1Reg)
-                    {
-                        assert(targetReg != op2Reg);
-                        assert(targetReg != op3Reg);
-
-                        GetEmitter()->emitIns_Mov(INS_mov, emitTypeSize(node), targetReg, op1Reg, /* canSkip */ true);
-                    }
+                    assert(targetReg == op1Reg || targetReg != op2Reg);
```

Comment:
Nit: We usually wrap the parts of compound boolean statements with parentheses, like so:
```suggestion
                    assert((targetReg == op1Reg) || (targetReg != op2Reg));
```

etc.

Reply:
Fixed!